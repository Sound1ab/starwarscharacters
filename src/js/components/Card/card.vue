<template>
	<div class="card">
		<h1 class="alpha">{{name}}</h1>
		<h2 class="beta">{{gender}}</h2>
		<table class="card__table">
			<tr
				v-for="(attribute, key) in $props"
				v-if="key !== 'name' && key !== 'gender'"
			>
				<td class="card__table-cell">{{key}}</td>
				<td class="card__table-cell">{{attribute}}</td>
			</tr>
		</table>
	</div>
</template>

<script>
	import VueTypes from 'vue-types';
	export default {
		name: 'card',
		props: {
			name: VueTypes.string.def(''),
			gender: VueTypes.string.def(''),
			hairColor: VueTypes.string.def(''),
			height: VueTypes.string.def(''),
			mass: VueTypes.string.def(''),
			skinColor: VueTypes.string.def(''),
			eyeColor: VueTypes.string.def(''),
			index: VueTypes.number.def(0)
		}
	};
</script>

<style lang="scss" type="text/scss">
	.card {
		width: 100%;
		height: em(328);
		padding: em(8);
		background-color: white;
		border-radius: 5px;
		box-shadow: 0 1px 4px 0 rgba(0,0,0,.2);
		transition: all .5s;
		&:hover {
			box-shadow: 0 0 20px 4px rgba(0,0,0,.1);
		}
		&__table {
			width: 100%;
		}
		&__table-cell {
			padding: em(8) 0;
		}
	}
</style>
