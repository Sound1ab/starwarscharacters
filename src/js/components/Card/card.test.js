/* eslint-disable */
import {shallow} from '@vue/test-utils';
import {createRenderer} from 'vue-server-renderer';
import Card from './card';

let mountComponent = () => shallow(Card);

describe('Card', () => {
	let wrapper;
	beforeEach(() => {
		wrapper = mountComponent();
		jest.clearAllMocks();
	});
	afterEach(() => {
		wrapper.destroy();
	});
	describe('htmlStructure', () => {
		it('html snapshot matches', () => {
			expect(wrapper.html()).toMatchSnapshot();
		});
	});
	describe('component', () => {
		it('vm - is instantiated', () => {
			expect(wrapper.isVueInstance()).toBe(true);
		});
	});
});
