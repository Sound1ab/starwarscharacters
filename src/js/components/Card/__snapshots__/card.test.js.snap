// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card htmlStructure html snapshot matches 1`] = `
<div class="card">
    <h1 class="alpha"></h1>
    <h2 class="beta"></h2>
    <table class="card__table">
        <!---->
        <!---->
        <tr>
            <td class="card__table-cell">hairColor</td>
            <td class="card__table-cell"></td>
        </tr>
        <tr>
            <td class="card__table-cell">height</td>
            <td class="card__table-cell"></td>
        </tr>
        <tr>
            <td class="card__table-cell">mass</td>
            <td class="card__table-cell"></td>
        </tr>
        <tr>
            <td class="card__table-cell">skinColor</td>
            <td class="card__table-cell"></td>
        </tr>
        <tr>
            <td class="card__table-cell">eyeColor</td>
            <td class="card__table-cell"></td>
        </tr>
        <tr>
            <td class="card__table-cell">index</td>
            <td class="card__table-cell">0</td>
        </tr>
    </table>
</div>
`;
